src/harness/reference_models/aggregate_interference/aggregate_interference.py

```diff
@@ -0,0 +1,362 @@
+  elif(fss_low_freq >= 3700 and fss_high_freq <= 4200) and\
+                                (fss_ttc_flag is True):
+    # 5MHz channelization is not required for Blocking protection
+    # FSS TT&C Blocking Algorithm
+    aggregate_interference = aggrFSSBlockingConstraint(
+      protection_point, interf.CBRS_LOW_FREQ_HZ, fss_low_freq, 
+      grant_list, fss_info, interf.ProtectedEntityType.FSS_BLOCKING)
+
+  elif(fss_low_freq >= 3700 and fss_high_freq <= 4200) and\
+                             (fss_ttc_flag is False):
+    logging.debug('Aggregate interference not calculated for FSS Pass band'
+    '3700 to 4200 and TT&C flag set to false')
+  return aggregate_interference
+
+
+def calculateAggregateInterferenceForEsc(esc_record, cbsd_list):
+  """Calculates aggregate interference for ESC.
+  
+  Calculates aggregate interference for each protection constraint in the ESC
+  based on the interferences caused by CBSD's grants in the neighborhood of
+  the ESC protection entity.
+  
+  Args:
+    esc_record: ESC protection entity
+    cbsd_list: list of CBSD objects containing RegistrationRequests and GrantRequests
+  Returns:
+    Aggregate interference to ESC in the dictionary format,where the key is a tuple. 
+    {(latitude, longitude, low_frequency, high_frequency,entity_type):
+    aggregate_interference(dBm)}
+  """
+
+  grant_list = interf.getAllGrantInformationFromCbsdDataDump(cbsd_list)
+
+  # Get the protection point of the ESC
+  esc_point = esc_record['installationParam']
+  protection_point = (esc_point['latitude'], esc_point['longitude'])
+
+  # Get azimuth radiation pattern
+  ant_pattern = esc_point['azimuthRadiationPattern']
+
+  # Initialize array of size length of ant_pattern list
+  ant_pattern_gain = np.arange(len(ant_pattern))
+  for i, pattern in enumerate(ant_pattern):
+    ant_pattern_gain[i] = pattern['gain']
+
+  # Get ESC antenna information
+  esc_antenna_info = interf.EscInformation(antenna_height=esc_point['height'],
+                       antenna_azimuth=esc_point['antennaAzimuth'],
+                       antenna_gain=esc_point['antennaGain'],
+                       antenna_pattern_gain=ant_pattern_gain)
+
+  # ESC Category A CBSDs between 3550 - 3660 MHz within 40 km
+  protection_channels = interf.getProtectedChannels(interf.ESC_CAT_A_LOW_FREQ_HZ, 
+                          interf.ESC_CAT_A_HIGH_FREQ_HZ)
+  aggregate_interference = aggregateInterferenceForPoint(protection_point, 
+                             protection_channels, interf.ESC_CAT_A_LOW_FREQ_HZ, 
+                             interf.ESC_CAT_A_HIGH_FREQ_HZ, grant_list, None, 
+                             esc_antenna_info, interf.ProtectedEntityType.ESC_CAT_A)
+  # ESC Category B CBSDs between 3550 - 3680 MHz within 80 km
+  protection_channels = interf.getProtectedChannels(interf.ESC_CAT_B_LOW_FREQ_HZ, 
+                          interf.ESC_CAT_B_HIGH_FREQ_HZ)
+  aggregate_interference = aggregateInterferenceForPoint(protection_point, 
```

Comment:
error. you are reassigning the variable aggregate_interferemce

What you probably intent is  
```
aggregate_interference.update(aggregateInterferenceForPomt(...))
```

Reply:
Not   required as aggregateInterference API is called per protected entity